torch/csrc/utils/pybind.h

```diff
@@ -1,33 +1,44 @@
 #pragma once
 
 #include <Python.h>
+
 #include <ATen/ATen.h>
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 #include "torch/csrc/DynamicTypes.h"
 #include "torch/csrc/autograd/python_variable.h"
 
+#include <stdexcept>
+
 namespace py = pybind11;
 
 namespace pybind11 { namespace detail {
 
-// handle Tensor <-> at::Tensor conversions
-// Python Variables are unpacked into Tensors
-template <> struct type_caster<at::Tensor> {
-public:
+// torch.autograd.Variable <-> at::Tensor conversions (without unwrapping)
+template <>
+struct type_caster<at::Tensor> {
+ public:
   PYBIND11_TYPE_CASTER(at::Tensor, _("at::Tensor"));
 
   bool load(handle src, bool) {
     PyObject* obj = src.ptr();
     if (THPVariable_Check(obj)) {
-      value = ((THPVariable*)obj)->cdata.data();
+      value = reinterpret_cast<THPVariable*>(obj)->cdata;
       return true;
     }
     return false;
   }
-  static handle cast(at::Tensor src, return_value_policy /* policy */, handle /* parent */) {
-    return handle(THPVariable_Wrap(torch::autograd::make_variable(src, false)));
+
+  static handle
+  cast(at::Tensor src, return_value_policy /* policy */, handle /* parent */) {
+#ifdef DEBUG
```

Comment:
Related to @apaszke's points about memory corruption, perhaps we should remove the DEBUG guards. It's probably worth always checking the type here, especially if we udpate & merge @ezyang's PR that improves the speed of "is_variable" checking.